README.md

```diff
@@ -3,9 +3,11 @@
 Formerly known as "Help to Heat".
 
 ## Initial Setup
-
-1. From the project root, run `cp envs/web.template envs/web`
-2. Populate the OS API key into the OS_API_KEY variable in `envs/web`. The key can be found in Keeper. Please only use a single value (i.e. `["my_key_here"]`), rather than all of the keys that are stored in Keeper, in order to avoid needing to rotate all of the keys if your local environment is accidentally leaked.
+1. [Install Poetry](https://python-poetry.org/docs/) on your machine
+2. [Install Python 3.9.13](https://www.python.org/downloads/release/python-3913/)
+3. Set up a virtual environment within the project for Poetry to manage your dependencies. A guide for Pycharm can be found [here](https://www.jetbrains.com/help/pycharm/poetry.html).
```

Comment:
Oh looks like my comment went missing 😢 meant to write here before

I don't think we need this given everything's running in a container? I'd hope not anyway - I don't particularly want to require people to install an old version of Python.

Reply:
Is there a reason to not do our dev work on the same version for compatibility, so our code runs how we expect it to?

Additionally, to use poetry, we need a compatible version of python installed with respect to the toml. Therefore to work with poetry on newer python versions, we'd have to loosen the requirement in the toml instead of locking it down as much as I've done currently.

Reply:
Yes, we do want to do our dev work on the same version of everything, but that's what Docker gives us. It should be the case that when you run the site/tests/whatever locally you're using Docker to do that - is that not true in practice?

Reply:
That's true, yes, while we're running GBIS locally, as the dockerfile has the version of python to be used in the image at the top of each file (which currently means it runs on 3.9.17). 
Does that mean we're good to loosen up the restriction in the .toml to something like "~3.9", so we can have our local version of python be more recent? Because our local version needs to be compatible with the restriction we've declared in the .toml to actually use Poetry. 

I can't find in the docs if poetry would complain that we've installed a dependency that is incompatible with 3.9, but works with 3.10 for example, and it wouldn't become clear until it tries to run in docker? Would it complain that the new dependency might not work with the potential minimum python version?

Reply:
I don't think you need to have a local version of python at all: it's in the container you spin up and everything runs in there.

I don't know if we even need to have Python in `pyproject.toml` at all? I'd kind of expect it to be implicit...

Reply:
We need a local version of python so our IDE can know what packages are installed.

To be clear, this doesn't affect running tests or the site. Its more a question of us needing it to perform dev work in the IDE prior to running anything. This local version we're using to perform dev work must be compatible with the toml file's specified range.

If I remove the python requirement entirely from the toml, I end up with issues surrounding the resolution of package versions. 
![image](https://github.com/UKGovernmentBEIS/help-to-heat-GBIS/assets/152421975/0f6e97a0-ac2d-406e-81e5-5a068736cccc)
I've consulted with Sammy, and we think we have to specify it in the toml to resolve this.

Reply:
I've made an update now to note in the poetry section's header that it is optional :)